framework/src/loops/ComputeInitialConditionThread.C

```diff
@@ -93,8 +93,26 @@ ComputeInitialConditionThread::operator()(const ConstElemRange & range)
             e, "Cyclic dependency detected in object ordering");
       }
 
-      for (auto ic : vec)
-        ic->compute();
+      /* Logic for running only old state initial conditions if they exist, but
+          running ICs normally if the old state ICs don't exist.
+      */
+      bool break_loop = false;
+      int loop_state;
+      for (loop_state = _fe_problem.global_current_state; loop_state >= 0; loop_state--)
+      {
+        for (auto ic : vec)
+        {
+          if (ic->_my_state == loop_state)
+          {
+            ic->compute();
+            break_loop = true;
+          }
+        }
+        if (break_loop == true)
+        {
+          break;
+        }
```

Comment:
```suggestion
        for (auto ic : vec)
          if (ic->_my_state == loop_state)
          {
            ic->compute();
            break_loop = true;
          }

        if (break_loop == true)
          break;
```

I'm not sure i understand the `break_loop` logic. Wouldn't that skip states for all ICs if one of them affects an older state? Even if they act on different variables?

Reply:
This for-loop logic was implemented inside an existing loop that runs for each relevant variable. So the ics being considered are all for the same variable. Does that make sense? check lines 78-81.